pkg/operator/aws_platform_service_location/controller.go

```diff
@@ -0,0 +1,71 @@
+package aws_platform_service_location
+
+import (
+	"context"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+type AWSPlatformServiceLocationController struct {
+	operatorClient operatorv1helpers.OperatorClient
+	infraClient    configv1client.InfrastructuresGetter
+}
+
+func NewController(operatorClient operatorv1helpers.OperatorClient, infraClient configv1client.InfrastructuresGetter, infraInformer cache.SharedIndexInformer, recorder events.Recorder) factory.Controller {
+	c := &AWSPlatformServiceLocationController{
+		operatorClient: operatorClient,
+		infraClient:    infraClient,
+	}
+	return factory.New().
+		WithInformers(
+			operatorClient.Informer(),
+			infraInformer,
+		).
+		WithSync(c.sync).
+		ResyncEvery(time.Minute).
+		ToController("AWSPlatformServiceLocationController", recorder)
+}
+
+func (c AWSPlatformServiceLocationController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	currentInfra, err := c.infraClient.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
```

Comment:
call out from here to a second function that does the work, like https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/operator/clustermembercontroller/clustermembercontroller.go#L125-L147 .  this allows all the working code to return errors as you expect, but always set status.

Reply:
@deads2k i wonder if a helper inside factory can be useful and if we want to promote this pattern further.

Reply:
@deads2k follow up: https://github.com/openshift/library-go/pull/757 :-)